### PR TITLE
Null safety

### PR DESCRIPTION
**Pull request by lielb100**
### Upgrade to Flutter SDK 2.12.0 sound null safety
I spent a lot of time and made the whole repo null safe, to make it more modern and future-proof.
Also I linted all the code with these analysis rules:
```yaml
  rules:
    file_names: false
    constant_identifier_names: false
    avoid_print: false
```
Suggesting to add a new branch for null_safety, and merge it when it is 100% stable.

---------------------------------------------------------------------------------------------
### Packages
- \- removed ```flutter_link_preview``` didn't have a null safe version so I replaced it
+ \+ added ```flutter_link_previewer``` new link preview package
+ \+ added ```flutter_chat_types``` for ```PreviewData``` type 
+ \+ added ```translator``` for tweet translation

### CODE REVIEW

**Comment by lielb100** [comment]
Sorry it talk long, but it's done and ready to be reviewd, I'd be happy to help with further devolopment of the app.

> Thanks @lielb100 for making this contribution. You have done a great job. Let me know if you are done with all required changes. Once you done I'll review the PR and let you know if something still needs to be change.

File: lib/ui/page/profile/EditProfilePage.dart [hunk 1]
```diff
@@ -69,7 +72,7 @@ class _EditProfilePageState extends State<EditProfilePage> {
           ),
         ),
         _entry('Name', controller: _name),
-        _entry('Bio', controller: _bio, maxLine: null),
+        _entry('Bio', controller: _bio),
```
Comment (TheAlphamerc) [inline lib/ui/page/profile/EditProfilePage.dart:75]: Padd null to `maxLine` to make multiple lines editor

**Review by TheAlphamerc — COMMENTED** [review]
@lielb100 I have added some comments plan check those.

File: lib/state/chats/chatState.dart [hunk 2]
```diff
@@ -82,11 +82,11 @@ class ChatState extends AppState {
   /// For more detail visit:- https://github.com/TheAlphamerc/flutter_twitter_clone/issues/28#issue-611695533
   /// For package detail check:-  https://pub.dev/packages/firebase_remote_config#-readme-tab-
   void getFCMServerKey() async {
-    final RemoteConfig remoteConfig = await RemoteConfig.instance;
+    final RemoteConfig remoteConfig = RemoteConfig.instance;
     await remoteConfig.fetchAndActivate();
     // await remoteConfig.
     var data = remoteConfig.getString('FcmServerKey');
-    if (data != null && data.isNotEmpty) {
+    if (data.isNotEmpty) {
```
[the remark this answers is not in the document]
Comment (lielb100) [reply lib/state/chats/chatState.dart:89]: Data is a non nullable String, I don't think it's possible for it to be null.

File: lib/resource/push_notification_service.dart [hunk 3]
```diff
@@ -60,21 +60,19 @@ class PushNotificationService {
...
-        FirebaseMessaging.onMessageOpenedApp.listen((event) {
-      if (event != null && event.data != null) {
-        if (event != null && event.data != null) {
-          try {
-            myBackgroundMessageHandler(event.data, onLaunch: true);
-          } catch (e) {
-            cprint(e, errorIn: "On onMessageOpenedApp");
-          }
+        FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage? event) {
+      if (event != null) {
```
[the remark this answers is not in the document]
Comment (lielb100) [reply lib/resource/push_notification_service.dart:64]: event.data is a non nullable String, I don't think it's possible for it to be null.

File: lib/ui/page/common/splash.dart [hunk 4]
```diff
@@ -89,12 +89,12 @@ class _SplashPageState extends State<SplashPage> {
...
   /// For package detail check:-  https://pub.dev/packages/firebase_remote_config#-readme-tab-
-  Future<String> _getAppVersionFromFirebaseConfig() async {
-    final RemoteConfig remoteConfig = await RemoteConfig.instance;
+  Future<String?> _getAppVersionFromFirebaseConfig() async {
+    final RemoteConfig remoteConfig = RemoteConfig.instance;
     await remoteConfig.fetchAndActivate();
     // await remoteConfig.activateFetched();
     var data = remoteConfig.getString('appVersion');
-    if (data != null && data.isNotEmpty) {
+    if (data.isNotEmpty) {
```
[the remark this answers is not in the document]
Comment (lielb100) [reply lib/ui/page/common/splash.dart:97]: Data is a non nullable String, I don't think it's possible for it to be null.

**Review by TheAlphamerc — APPROVED** [review]
Thanks @lielb100 for making this contribution.

**Pull request merged**